momento-sdk/src/main/java/momento/sdk/messages/CacheGetResponse.java

```diff
@@ -1,109 +1,80 @@
 package momento.sdk.messages;
 
 import com.google.protobuf.ByteString;
-import grpc.cache_client.ECacheResult;
 import java.io.InputStream;
 import java.nio.ByteBuffer;
 import java.nio.charset.Charset;
 import java.nio.charset.StandardCharsets;
-import java.util.Optional;
-import momento.sdk.exceptions.InternalServerException;
+import momento.sdk.exceptions.SdkException;
 
 /** Response for a cache get operation */
-public final class CacheGetResponse {
-  private final ByteString body;
-  private final CacheGetStatus status;
+public abstract class CacheGetResponse {
 
-  public CacheGetResponse(ECacheResult status, ByteString body) {
-    this.body = body;
-    this.status = convert(status);
-  }
+  public static class Hit extends CacheGetResponse {
+    private final ByteString body;
 
-  /**
-   * Determine the result of the Get operation.
-   *
-   * <p>Valid values are {@link CacheGetStatus#HIT} and {@link CacheGetStatus#MISS}.
-   *
-   * @return The result of Cache Get Operation
-   */
-  public CacheGetStatus status() {
-    return status;
-  }
+    public Hit(ByteString body) {
+      this.body = body;
+    }
 
-  /**
-   * Value stored in the cache as a byte array.
-   *
-   * @return Value stored for the given key. {@link Optional#empty()} if the lookup resulted in a
-   *     cache miss.
-   */
-  public Optional<byte[]> byteArray() {
-    if (status != CacheGetStatus.HIT) {
-      return Optional.empty();
+    /**
+     * Value stored in the cache as a byte array.
+     *
+     * @return Value stored for the given key.
+     */
+    public byte[] byteArray() {
```

Comment:
we've established a convention that all of our value accessors will have a name that starts with the word `value` so that it's easy to find them across different response types, and so that code completion will always group them together nicely.